java-cfenv-boot/src/main/java/io/pivotal/cfenv/spring/boot/GenAICfEnvProcessor.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.pivotal.cfenv.spring.boot;
+
+import java.util.Map;
+
+import io.pivotal.cfenv.core.CfCredentials;
+import io.pivotal.cfenv.core.CfService;
+
+/**
+ * Retrieve GenAI on Tanzu Platform properties from {@link CfCredentials} and
+ * set {@literal spring.ai}
+ * Boot properties.
+ *
+ * @author Stuart Charlton
+ **/
+public class GenAICfEnvProcessor implements CfEnvProcessor {
+
+	@Override
+	public boolean accept(CfService service) {
+		return service.existsByTagIgnoreCase("genai") ||
+				service.existsByLabelStartsWith("genai");
+	}
+
+	@Override
+	public void process(CfCredentials cfCredentials, Map<String, Object> properties) {
+		properties.put("spring.ai.openai.base-url", cfCredentials.getString("api_base"));
```

Comment:
Hello!
Thanks for your contribution!
Never used the GenAI tile before in CF / TAS.
With a quick glance at your PR, it looks like it's `openai` specific; while `spring-ai` supports at least Gemini and Ollama as well.(and others too)
Is it on purpose that the properties are just for OpenAI in this PR?
Thanks!

Reply:
Hi Anthony!

The GenAI tile exposes a virtual OpenAI API endpoint with a custom base URL so that the platform user gets a private OpenAI.com-like service that proxies private models hosted by VLLM, Ollama, VMware Private AI with NVIDIA NIM, etc. behind the scenes.    This is why we're only exposing OpenAI, you can (in theory) take any OpenAI-API client code and plug it into GenAI and it should work.

Reply:
oh OK!
that makes sense, thank you for the explanation!